core/src/main/scala/kafka/server/KafkaRequestHandler.scala

```diff
@@ -386,7 +431,26 @@ class BrokerTopicMetrics(name: Option[String], configOpt: java.util.Optional[Kaf
       meter.close()
   }
 
-  def close(): Unit = metricTypeMap.values.foreach(_.close())
+  def close(): Unit = {
+    metricTypeMap.values.foreach(_.close())
+    metricGaugeTypeMap.values.foreach(_.close())
+  }
+}
+
+class BrokerTopicAggregatedMetric() {
+  private val partitionMetricValues = new ConcurrentHashMap[Int, Long]().asScala
+
+  def setPartitionMetricValue(partition: Int, partitionValue: Long): Option[Long] = {
+    partitionMetricValues.put(partition, partitionValue)
+  }
+
+  def removePartition(partition: Int): Option[Long] = {
+    partitionMetricValues.remove(partition)
+  }
+
+  def value(): Long = partitionMetricValues.values.sum
```

Comment:
Also, instead of computing the aggregated `value` on each reporting (typically: 10 s). Can we keep one variable `value` and update it while recording the partition level metrics?

Reply:
I actually have a preference for not updating one variable. The reason I don't want to use one variable is that a concurrent map provides fine-grained controls out of the box that no one else will be modifying the value while we read it. If I use a value then I have to acquire a lock. What are your thoughts?